src/models/util.ts

```diff
@@ -0,0 +1,35 @@
+import * as Knex from 'knex';
+import { intersection, compose } from 'ramda';
+import { CommonQueryOptions } from './models';
+
+
+export const Utils: {
+  [k: string]: (...a: any[]) => (q: Knex.QueryBuilder) => Knex.QueryBuilder
+} = {
+  limit: (n: number) => (q: Knex.QueryBuilder) => q.limit(n),
+  offset: (n: number) => (q: Knex.QueryBuilder) => q.offset(n),
+  order: (c: string, d: string) => (q: Knex.QueryBuilder) => q.orderBy(c, d),
+};
+
+export const applyQueryModifiers = (p: Knex.QueryBuilder, opts: CommonQueryOptions) => {
+  const modifiers = intersection(Object.keys(opts), Object.keys(Utils));
+
+  const query = modifiers
+    .reduce((acc, k: keyof CommonQueryOptions) => {
```

Comment:
Out of curiosity, why are you grabbing the keys of `Utils` and `CommonQueryOptions` separately for the same data?

Reply:
`opts` may include zero or more of the keys in `CommonQueryOptions`, and `Utils` does not support `fields` (because it's dealt with in the `select`), so taking the intersection of the two means I will only iterate over keys actually in `opts` and actually supported by `Utils`.

I think that's what you were asking but am not sure.